client/src/lib/firebase/auth.ts

```diff
@@ -37,14 +42,81 @@ export async function loginAnonymous() {
 
 export function deleteAccount() {
   const user = auth.currentUser;
-
   if (user == null) {
     throw new Error("User is not defined.");
   }
 
   deleteUser(user);
 }
 
+export function linkWithGoogle() {
+  const user = auth.currentUser;
+  if (user != null) {
+    const signInMethods = user.providerData;
+
+    signInMethods.forEach((provider) => {
+      if (provider.providerId == "google.com") {
+        throw new Error("a-google-account-already-exists-for-this-user");
+      }
+    });
+
+    linkWithPopup(user, google);
+    return;
```

Comment:
probably want to return `linkWithPopup` then awaited when you use it

Reply:
probably want to return `linkWithPopup` then awaited when you use it

Reply:
fixed